src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -0,0 +1,77 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Strings;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileDocumentSynchronizationVetoer;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.redhat.devtools.intellij.common.utils.UIHelper;
+import com.redhat.devtools.intellij.tektoncd.utils.CRDHelper;
+import com.redhat.devtools.intellij.tektoncd.utils.JSONHelper;
+import com.redhat.devtools.intellij.tektoncd.utils.YAMLHelper;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.common.CommonConstants.KIND_PLURAL;
+
+public class SaveInEditorListener extends FileDocumentSynchronizationVetoer {
+    Logger logger = LoggerFactory.getLogger(SaveInEditorListener.class);
+
+    @Override
+    public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplicit) {
+        VirtualFile vf = FileDocumentManager.getInstance().getFile(document);
+        // if file is not related to tekton we can skip it
+        if (vf == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
+            return true;
+        }
+
+        String namespace, name;
+        JsonNode spec;
+        try {
+            namespace = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "namespace"});
+            name = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "name"});
+            if (Strings.isNullOrEmpty(namespace) || Strings.isNullOrEmpty(name)) {
+                throw new IOException("Tekton file has not a valid format. Namespace and/or name fields are invalid.");
+            }
+            spec = YAMLHelper.getValueFromYAML(document.getText(), new String[] {"spec"});
+            if (spec == null) {
+                throw new IOException("Tekton file has not a valid format. Spec field is not found.");
+            }
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Error"));
+            return false;
+        }
+
+        Notification notification;
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
+            CustomResourceDefinitionContext crdContext = CRDHelper.getCRDContext(vf.getUserData(KIND_PLURAL));
```

Comment:
We have an issue with versions are you are hard coding v1alpha1 where Tekton is already switching to v1alpha2 so I think we should extract the CRD context from the resource file.